Code monitoring docs (#16173)
sourcegraph/sourcegraph: doc/code_monitoring/explanations/core_concepts.md

Sourcegraph will run the search query perioidcally, and when new results for the query are detected, a trigger event is executed. In response to the trigger event, any _actions_ attached to the code monitor will be executed.

**Query requirements**

A query that is defined as a trigger must be a diff or commit search. In other words, the query must contain `type:commit` or `type:diff`. This allows Sourcegraph to detect new search results periodically.

## Actions

An _action_ is executed in response to a trigger event. Currently, the only action supported is sending a notification email to the owner of the code monitor.

In response to a trigger event, Sourcegraph will send an email containing a link to the newly detected results to the owner of the code monitor.

## Current flow

To put it all together, a code monitor defined today would have a flow similar to the following: 

A user creates a code monitor, which consists of: 
  * a name for the monitor
  * a trigger, which consists of a search query to run periodically,
  * and an action, which is sending an email to the user when new results appear

> Sourcegraph runs the query perioidcally. When new results are detected, an email will be sent to the user that created the monitor. The email contains a link to the newly detected search results.

Reviewer: ```suggestion
Sourcegraph runs the query perioidcally. When new results are detected, an email is sent to the user that created the monitor. The email contains a link to the newly detected search results.
```
